src/base/random.cc

```diff
@@ -104,7 +100,28 @@ void RNG::write_state(StreamWrite& sw) {
 	sw.unsigned_32(state1);
 }
 
+static RNG static_rng_(clock());
+uint32_t RNG::static_rand() {
+	return static_rng_.rand();
+}
+
 std::string generate_random_uuid() {
-	static boost::uuids::random_generator gen;
-	return boost::uuids::to_string(gen());
+	uint32_t values[4];
+	RNG temp_rng;
+	int64_t seed = clock();
+	for (uint32_t& val_ref : values) {
+		seed += std::chrono::time_point_cast<std::chrono::nanoseconds>(
+		           std::chrono::high_resolution_clock::now())
+		           .time_since_epoch()
+		           .count();
+		temp_rng.seed(seed % 0xfedcba98);
+		val_ref = temp_rng.rand();
+	}
+
+	char buffer[16 * 4 + 4 + 1];
+	snprintf(buffer, sizeof(buffer), "%04x%04x-%04x-%04x-%04x-%04x%04x%04x", values[0] & 0xffff,
+	         values[1] & 0xffff, values[2] & 0xffff, values[3] & 0xffff,
+	         (values[0] & 0xffff0000) >> 16, (values[1] & 0xffff0000) >> 16,
+	         (values[2] & 0xffff0000) >> 16, (values[3] & 0xffff0000) >> 16);
```

Comment:
Does this generate a UUID conforming to any [standard](https://pubs.opengroup.org/onlinepubs/9629399/apdxa.htm)? If so, please attach a reference.

Reply:
This doesn't follow a standard because we use UUIDs for only one purpose where the only property we need is that no two WL clients will generate the same UUID within a short time of each other:
https://github.com/widelands/widelands/blob/cbe7f895fdd9a4790a5cedfdc2c1ae216fd348a6/src/network/internet_gaming_protocol.h#L107-L128
Added a comment to the UUID generator function.